content/kotlin/concepts/strings/terms/filter/filter.md

```diff
@@ -0,0 +1,47 @@
+---
+Title: 'filter()'
+Description: 'filter elements from collection based on a given predicate.'
+Subjects:
+  - 'Computer Science'
+  - 'Web Development'
+Tags:
+  - 'Collections'
+  - 'Filter'
+  - 'Strings'
+CatalogContent:
+  - 'learn-kotlin'
+  - 'paths/computer-science'
+---
+
+## Syntax
+
+The **filter()** function is used to filter elements from a collection based on a given _predicate_. It returns a new collection containing only elements which statifies a specified condition.
+
+```pseudo
+str.filter(predicate: (char) -> Boolean)
+
+```
+
+Given a string `str` above, `filter()` method can filter the characters from the string using a predicate which is passed as an argument.
```

Comment:
```suggestion
- When working with the string `str` mentioned above, the `filter()` method allows selective removal of characters from the string based on a provided predicate.
```

Reply:
@dakshdeepHERE, `filter()` method does not remove the elements from the string instead it creates a new collection which includes only selective elements

Reply:
Yes let me refactor it. It basically create a new collection that includes only the elements that satisfy a specified predicate.

Reply:
yes, I have made the changes as you suggested also this one. You can check it on my latest commit.